jwk/jwk.go

```diff
@@ -744,3 +744,13 @@ func Equal(k1, k2 Key) bool {
 
 	return bytes.Equal(tp1, tp2)
 }
+
```

Comment:
We're planning to make it possible for third parties to add their new key types, so I think this interface should be public, allowing new keys to decide for themselves if they want to opt-in to this mechanism.

How about

```go
package jwk

type AsymmetricKey interface {
   IsPrivate() (bool, error)
}
```